stackslib/src/burnchains/mod.rs

```diff
@@ -595,6 +595,9 @@ impl PoxConstants {
 
             // NOTE: first block in reward cycle is mod 1, so mod 0 is the last block in the
             // prepare phase.
+            // TODO: I *think* the logic of `== 0` here requires some further digging.
+            //  `mod 0` may not have any rewards, but it does not behave like "prepare phase" blocks:
+            //  is it already a member of reward cycle "N" where N = block_height / reward_cycle_len
```

Comment:
I added this comment here for some visibility -- this may impact block inventory logic, right, @jcnelson?

The "0" reward index isn't really a prepare phase block for cycle N: it's a member of reward cycle N, not N-1.

Reply:
Changing this would impact far more than block inventory logic.  I'm afraid this is consensus-level logic.

Fundamentally, there's an unsoundness in the code for deciding whether or not a block height is in the prepare phase or reward phase.  In some places, like this one and others, a block whose height modulo the reward cycle length is 0 is in the prepare phase, since the reward phase begins when the block height modulo the reward cycle length is 1 or greater.  In other places, like `static_block_height_to_reward_cycle`, a block is in the reward phase if its height modulo the reward cycle length is 0.

Unfortunately, this code here (as well as the code paths that declare the start of a reward cycle to be at each block whose heights modulo the reward cycle length is 1) is used all over the place in the blockchain.  We can't change it.

Reply:
This is also different from `is-in-prepare-phase` in the voting contract.

Reply:
Also `reward-cycle-to-burn-height` in the voting contract is inconsistent with the code here.